### PR TITLE
New version scheme

### PR DESCRIPTION
**Pull request by nerzhul**
I pushed the two result commits for a release as a proof of concept

Here is the console log

```
> ./util/bump_version.sh 
Current Minetest version: 0.5.0
Current Android version code: 17
Set android version code [18]:

New android version code: 18
[new_version_scheme bac8b86e7] Bump version to 0.5.0
 2 files changed, 2 insertions(+), 2 deletions(-)
Tagging 0.5.0
Set next major [0]:
Set next minor [5]:
Set next patch [1]:

New version: 0.5.1
Creating "return back to development" commit
[new_version_scheme f7ce591c6] Continue with 0.5.1-dev
 4 files changed, 5 insertions(+), 5 deletions(-)
```

### CODE REVIEW

File: util/bump_version.sh [hunk 1]
```diff
@@ -16,7 +16,50 @@ prompt_for_number() {
...
+
+	sed -i -re "s/^set\(VERSION_MINOR [0-9]+\)$/set(VERSION_MINOR $NEXT_VERSION_MINOR)/" CMakeLists.txt || die "Failed to update VERSION_MINOR"
+
+	sed -i -re "s/^set\(VERSION_PATCH [0-9]+\)$/set(VERSION_PATCH $NEXT_VERSION_PATCH)/" CMakeLists.txt || die "Failed to update VERSION_PATCH"
+
+	sed -i -re "1s/[0-9]+\.[0-9]+\.[0-9]+/$NEXT_VERSION/g" doc/lua_api.txt || die "Failed to update doc/lua_api.txt"
+
+	sed -i -re "1s/[0-9]+\.[0-9]+\.[0-9]+/$NEXT_VERSION/g" doc/menu_lua_api.txt || die "Failed to update doc/menu_lua_api.txt"
+
+	sed -i -re "1s/[0-9]+\.[0-9]+\.[0-9]+/$NEXT_VERSION/g" doc/client_lua_api.md || die "Failed to update doc/client_lua_api.md"
```
Comment (SmallJoker) [inline util/bump_version.sh]: Any chance to split these on two lines?

Comment (nerzhul) [inline util/bump_version.sh]: what do you mean ? it's the minetest current version and we update the doc when we develop

Comment (SmallJoker) [inline util/bump_version.sh]: > 90 characters long, that's what I mean.

**Comment by rubenwardy** [comment]
I suggest adding a new header in readme about versions.

> Minetest doesn't follow semvar. Instead, we do something roughly similar to 0.major.minor.
>
> Since 0.5.0-dev and 0.4.17-dev, the dev notation refers to the next release, ie: 0.5.0-dev is the development version leading to 0.5.0.
> Prior to that, we used oldversion-dev.

I'm on my phone, so I can't type that much. May be worth expanding on that

File: util/bump_version.sh (hunk 1 above)
Comment (nerzhul) [inline util/bump_version.sh]: the problem when splitting a such line in shell is making the command unreadable :s

**Comment by nerzhul** [comment]
Approved by @sfan5 on IRC http://irc.minetest.net/minetest-dev/2017-08-20#i_5049517

**Pull request merged**